### PR TITLE
Use Angle Bracket Notation (Fixes #64)

### LINKED ISSUE #64: renaming keys to vim nomenclature
Vim uses names like `<cr>` (`return`), `<C-m>` (`ctrl+m`). This will be required later to process keys, for example for the `:normal` command.

@jpoon Thoughts?

### PR DESCRIPTION
**Pull request by jpoon**
Easing myself back into the project after a short hiatus with this simple change. 

Update: Translating key presses to Vim's angle bracket notation:

`ctrl+*` becomes `<C-*>`
`escape` becomes `<Esc>`
`backspace` becomes `<BS>`
`delete` becomes `<Del>`

### CODE REVIEW

**Comment by johnfn** [comment]
Could you actually fix this in the more general way by obliterating ctrl+* from the codebase entirely and replacing all such instances with <c-*>? 

It would fix our longest standing bug #64 :)

**Comment by jpoon** [comment]
Argh, just realized this will break backwards compatibility for those that have used a `otherModesKeyBindings` settings.

**Comment by johnfn** [comment]
True, but we don't actually document how to do that anywhere (intentionally, since I knew this was gonna happen eventually...)

Ideally we could just show an error to people who we detect have ctrl+* keybindings - say "you need to update your old keybindings to match the new format"

We could even handle both in the keybindings parser.

**Comment by jpoon** [comment]
That should do it. Moved the angle bracket translation logic to util.ts where it is now called by:

1) `extension.ts` where the vscode key presses come in
2) `remapper.ts` to remap keys for insertModeKeyBindings and otherModesKeyBindings (ie. we have backwards compat)

File: README.md [hunk 1]
```diff
@@ -42,26 +42,26 @@ On OS X, open Terminal and run the following command:
...
+#### How can I bind `jj` to `<Esc>`?
 
 1. Add the following to `settings.json` (open the Command Pallete and search for "User Settings"):
 
    ```
       "vim.insertModeKeyBindings": [
            {
                "before": ["j", "j"],
-               "after": ["<escape>"]
+               "after": ["<Esc>"]
```
Comment (johnfn) [inline README.md]: Can we make this `<esc>` to stay in line with Vim?

Comment (jpoon) [inline README.md]: Are you referring to the capitalization? [VimDocs](http://vimdoc.sourceforge.net/htmldoc/intro.html#<Esc>) have it as a capital `E` so following suit.

Comment (johnfn) [inline README.md]: I didn't know that! Okay, that's fine.

File: src/actions/actions.ts [hunk 2]
```diff
@@ -413,7 +413,7 @@ class CommandEsc extends BaseCommand {
     ModeName.SearchInProgressMode,
     ModeName.Replace
   ];
-  keys = ["<escape>"];
+  keys = ["<Esc>"];
```
Comment (johnfn) [inline src/actions/actions.ts]: This must have been annoying to do. Thanks.

I would love to extract these out into a string enum at some point.

**Pull request merged**